core/trino-spi/src/main/java/io/trino/spi/eventlistener/RoutineInfo.java

```diff
@@ -25,14 +25,23 @@
 {
     private final String routine;
     private final String authorization;
+    private final ClauseInfo clauseInfo;
+
+    @Deprecated
+    public RoutineInfo(String routine, String authorization)
+    {
+        this(routine, authorization, ClauseInfo.UNSPECIFIED);
+    }
 
     @JsonCreator
     public RoutineInfo(
             @JsonProperty("routine") String routine,
-            @JsonProperty("authorization") String authorization)
+            @JsonProperty("authorization") String authorization,
+            @JsonProperty("clauseInfo") ClauseInfo clauseInfo)
```

Comment:
I do think that RoutineInfo should be meant for consumption by the plugins, so ctor changes should be excempt for backward compat checks. However, we also annotated the class with JSON annotations, so we made JSON form part of the contract.

If we take backward compat seriously here, we need to allow this class to deserialize previous forms, those missing `clauseInfo`.
Therefore make `clauseInfo` an Optional field.

Reply:
The RoutineInfo is part of QueryCompletedEvent->QueryMetadata which can be used by plugins

Reply:
> make `clauseInfo` an Optional field.

Done